packages/gafl-webapp-service/src/pages/multibuy/add-licence/__tests__/route.spec.js

```diff
@@ -0,0 +1,51 @@
+import { getData } from '../route.js'
+import { CONTACT_SUMMARY, LICENCE_SUMMARY } from '../../../../uri.js'
+
+import GetDataRedirect from '../../../../handlers/get-data-redirect.js'
+jest.mock('../../../../handlers/get-data-redirect.js')
+
+import { createMockRequest } from '../../../../__mocks__/request.js'
+
+describe('The add licence route .getData', () => {
+  it('redirects to the licence summary page if the user has not visited it yet', async () => {
+    const mockRequest = createMockRequest()
+    try {
+      await getData(mockRequest)
+    } catch (err) {
+      expect(GetDataRedirect).toBeCalledWith(LICENCE_SUMMARY.uri)
+    }
```

Comment:
There's a bit of a trap here, which can exist for tests that throw errors. The problem with the way around you've done this is that if the `getData` function doesn't throw, this test will seemingly pass (in fact, the expect doesn't end up being called, as no error is thrown, so it's just skipped over). If you change the start of `getData` as follows:

```
export const getData = async request => {
  const status = await request.cache().helpers.status.getCurrentPermission()

  if (!status[LICENCE_SUMMARY.page]) {
    // throw new GetDataRedirect(LICENCE_SUMMARY.uri)
  }

  if (!status[CONTACT_SUMMARY.page]) {
    // throw new GetDataRedirect(CONTACT_SUMMARY.uri)
  }

  GetDataRedirect(CONTACT_SUMMARY.uri)
  
  // rest of function as it was...
}
```
All tests still 'pass'.

To overcome this, I think we need to do a couple of things. First of all, we need to modify GetDataRedirect to pass the redirectUrl to error as a message:

```
/**
 * This can be thrown by the get data functions to force a redirect.
 * If specific data items necessary to display the page have not been set
 */

export default class GetDataRedirect extends Error {
  constructor (redirectUrl) {
    super(redirectUrl) // change is here
    this.redirectUrl = redirectUrl
  }
}
```
Then, we can test by doing the following:
```
  it('redirects to the licence summary page if the user has not visited it yet', async () => {
    const mockRequest = createMockRequest()
    await expect(() => getData(mockRequest)).rejects.toThrow(LICENCE_SUMMARY.uri)
  })
```
Could I leave you to modify the other test?